app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/blocklist/AppTPBlockListInterceptorApiPluginTest.kt

```diff
@@ -0,0 +1,267 @@
+import com.duckduckgo.data.store.api.SharedPreferencesProvider
+import com.duckduckgo.di.DaggerSet
+import com.duckduckgo.feature.toggles.api.FakeToggleStore
+import com.duckduckgo.feature.toggles.api.FeatureToggles
+import com.duckduckgo.feature.toggles.api.FeatureTogglesInventory
+import com.duckduckgo.feature.toggles.api.Toggle
+import com.duckduckgo.feature.toggles.api.Toggle.DefaultValue
+import com.duckduckgo.feature.toggles.api.Toggle.State
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.CONTROL
+import com.duckduckgo.mobile.android.vpn.feature.AppTpRemoteFeatures.Cohorts.TREATMENT
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixelNames
+import com.duckduckgo.mobile.android.vpn.pixels.DeviceShieldPixels
+import com.duckduckgo.mobile.android.vpn.pixels.RealDeviceShieldPixels
+import com.squareup.moshi.Moshi
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+
+// TODO: currently copied from feature-toggles-impl, but might be a better way to share this (2025-04-02)
+class RealFeatureTogglesInventory @Inject constructor(
+    private val toggles: DaggerSet<FeatureTogglesInventory>,
+    private val dispatcherProvider: DispatcherProvider,
+) : FeatureTogglesInventory {
+    override suspend fun getAll(): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext toggles.flatMap { it.getAll() }.distinctBy { it.featureName() }
+    }
+
+    override suspend fun getAllTogglesForParent(name: String): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext getAll().filter { it.featureName().parentName == name }
+    }
+
+    override suspend fun getAllActiveExperimentToggles(): List<Toggle> = withContext(dispatcherProvider.io()) {
+        return@withContext getAll().filter { it.getCohort() != null && it.isEnabled() }
+    }
+}
+
+@SuppressLint("DenyListedApi")
+class AppTPBlockListInterceptorApiPluginTest {
+
+    @get:Rule
+    @Suppress("unused")
+    val coroutineRule = CoroutineTestRule()
+
+    private lateinit var testBlockListFeature: TestBlockListFeature
+    private lateinit var inventory: FeatureTogglesInventory
+    private lateinit var interceptor: AppTPBlockListInterceptorApiPlugin
+    private val moshi = Moshi.Builder().build()
+
+    private data class Config(
+        val treatmentUrl: String? = null,
+        val controlUrl: String? = null,
+        val nextUrl: String? = null,
+    )
+    private val configAdapter = moshi.adapter(Config::class.java)
+
+    private val pixel = mock<Pixel>()
+    private val sharedPreferencesProvider = mock<SharedPreferencesProvider>()
+    private val prefs = InMemorySharedPreferences()
```

Comment:
Use FakeSharedPreferencesProvider from `data-store-test` instead. You'll need to `testImplementation` import `data-store-test`